rcl/test/rcl/test_graph.cpp

```diff
@@ -150,42 +150,6 @@ TEST_F(
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
 }
 
-/* Test the rcl_get_topic_names_and_types and rcl_destroy_topic_names_and_types functions.
- *
- * This does not test content of the rcl_topic_names_and_types_t structure.
- */
-TEST_F(
-  CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
-  test_rcl_get_and_destroy_node_names) {
-  stop_memory_checking();
-  rcl_ret_t ret;
-  rcl_string_array_t node_names {};
-  rcl_node_t zero_node = rcl_get_zero_initialized_node();
-  // invalid node
-  ret = rcl_get_node_names(nullptr, &node_names);
-  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string_safe();
-  rcl_reset_error();
-  ret = rcl_get_node_names(&zero_node, &node_names);
-  EXPECT_EQ(RCL_RET_NODE_INVALID, ret) << rcl_get_error_string_safe();
-  rcl_reset_error();
-  ret = rcl_get_node_names(this->old_node_ptr, &node_names);
-  EXPECT_EQ(RCL_RET_NODE_INVALID, ret) << rcl_get_error_string_safe();
-  rcl_reset_error();
-  // invalid topic_names_and_types
-  ret = rcl_get_node_names(this->node_ptr, nullptr);
-  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string_safe();
-  rcl_reset_error();
-  // invalid argument to rcl_destroy_topic_names_and_types
-  ret = rcl_destroy_node_names(nullptr);
-  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string_safe();
-  rcl_reset_error();
-  // valid calls
-  ret = rcl_get_node_names(this->node_ptr, &node_names);
-  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
-  ret = rcl_destroy_node_names(&node_names);
-  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
-}
-
```

Comment:
I'll removed them because of duplication.

The create and fini functions are covered in this one: https://github.com/ros2/c_utilities/blob/master/test/test_string_array.cpp
The semantics of `get_node_names` is covered here:
https://github.com/ros2/rcl/blob/master/rcl/test/rcl/test_get_node_names.cpp

Are we missing any tests on that?